packages/loot-core/src/server/main.ts

```diff
@@ -1892,13 +1904,103 @@ handlers['delete-budget'] = async function ({ id, cloudFileId }) {
 
   // If a local file exists, you can delete it by passing its local id
   if (id) {
+    // loading and then closing the budget is a hack to be able to delete
+    // the budget file if it hasn't been opened yet.  This needs a better
+    // way, but works for now.
+    await loadBudget(id);
+    await handlers['close-budget']();
+
     const budgetDir = fs.getBudgetDir(id);
     await fs.removeDirRecursively(budgetDir);
   }
 
   return 'ok';
 };
 
+handlers['duplicate-budget'] = async function ({
+  id,
+  newName,
+  cloudSync,
+  open,
+}): Promise<string> {
+  if (!id) throw new Error('Unable to duplicate a budget that is not local.');
+
+  const { valid, message } = await validateBudgetName(newName);
+  if (!valid) throw new Error(message);
+
+  const budgetDir = fs.getBudgetDir(id);
+
+  let budgetFileName = newName;
+  if (budgetFileName.indexOf(' - copy') !== -1) {
```

Comment:
This is hardcoding english in budget names

Reply:
Not sure how to use translate on the loot-core server side.  ```useTranslate``` is not on the server side and when I try the ```t``` function from ```i18next``` it returns undefined.  There is a small discussion on Discord about it not working on the server side.  If you have any information, I'd be glad to look into it.

Reply:
If there's no way to translate the loot-core server, then I can remove these couple of lines.  They do nothing to the function, just make the filename prettier.